bciers/apps/administration/app/data/jsonSchema/facilitiesLfo.ts

```diff
@@ -23,46 +23,122 @@ const section1: RJSFSchema = {
-    is_current_year: {
-      type: "boolean",
-      title: `Did this facility begin operations in ${
-        currentYear - 1
-      } or ${currentYear}?`,
-      default: false,
-    },
-    bcghg_id: { type: "string", title: "BCGHGID" },
   },
-  allOf: [
-    {
-      if: {
-        properties: {
-          is_current_year: {
-            const: true,
+  dependencies: {
+    type: {
+      oneOf: [
+        {
+          properties: {
+            type: {
+              const: "Large Facility",
+            },
+            well_authorization_numbers: {
+              type: "array",
+              items: {
+                type: "number",
+              },
+              title: "BC Energy Regulator Well Authorization Number(s)",
+            },
+            is_current_year: {
+              type: "boolean",
+              title: `Did this facility begin operations in ${
+                currentYear - 1
+              } or ${currentYear}?`,
+              default: false,
+            },
+            bcghg_id: { type: "string", title: "BCGHGID" },
+          },
+          dependencies: {
+            is_current_year: {
+              oneOf: [
+                {
+                  properties: {
+                    is_current_year: {
+                      const: false,
+                    },
+                  },
+                },
+                {
+                  properties: {
+                    is_current_year: {
+                      const: true,
+                    },
+                    starting_date: {
+                      type: "string",
+                      title: "Date of facility starting operations:",
+                      allOf: [
+                        { format: "date_format" },
+                        { format: "starting_date_year" },
+                      ],
+                      required: ["starting_date"],
+                    },
+                  },
+                },
+              ],
+            },
           },
         },
-      },
-      then: {
-        properties: {
-          starting_date: {
-            type: "string",
-            title: "Date of facility starting operations:",
-            allOf: [
-              { format: "date_format" },
-              { format: "starting_date_year" },
-            ],
+        {
+          properties: {
+            type: {
+              const: "Medium Facility",
+            },
+            well_authorization_numbers: {
+              type: "array",
+              items: {
+                type: "number",
+              },
+              title: "BC Energy Regulator Well Authorization Number(s)",
+            },
+            is_current_year: {
+              type: "boolean",
+              title: `Did this facility begin operations in ${
+                currentYear - 1
+              } or ${currentYear}?`,
+              default: false,
+            },
+            bcghg_id: { type: "string", title: "BCGHGID" },
+          },
+          dependencies: {
```

Comment:
I tried putting this in a variable and reusing it for both large and medium facilities, but I hit type errors